docs: Introduce spec overview and template (#14839)
influxdata/telegraf: docs/specs/overview.md

Reviewer: Maybe rename this to README.md so GitHub directly opens this one when opening the specs dir?

Author: agreed - done

> # Telegraf Spec Overview

Reviewer 2: ```suggestion
# Telegraf Specification Overview
```

## Objective

> Define and layout the Telegraf spec process.

Reviewer 2: ```suggestion
Define and layout the Telegraf specification process.
```

## Overview

The general goal of a spec is to detail the work that needs to get accomplished
for a new feature. A developer should be able to pick up a spec and have a
decent understanding of the objective, the steps required, and most of the
general design decisions.

> The specs can then live in the Telegraf repo to share and involve with the
> community the planning process that goes into larger changes and have a public
> historical record for changes.

Reviewer 2: ```suggestion
The specs can then live in the Telegraf repository to share and involve with the
community in the process of planning larger changes or new features. The specs
also serve as a public historical record for changes.
```
